packages/components/src/components/FileUploader/FileUploader.test.tsx

```diff
@@ -127,6 +127,39 @@ describe("<FileUploader />", () => {
     });
   });
 
+  describe("when there is an error", () => {
+    it("renders Upload button but not Remove button when there is an errorMessage", () => {
+      renderFileUploader({
+        label: "Permit",
+        errorMessage: "You must upload a file",
+      });
+
+      expect(screen.getByText("Permit")).toBeInTheDocument();
+      expect(
+        screen.getByRole("button", { name: "Upload" })
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByRole("button", { name: "Remove file" })
+      ).not.toBeInTheDocument();
+      expect(screen.getByRole("alert")).toHaveTextContent(
+        "You must upload a file"
+      );
+    });
+
+    it("renders Upload button but not alert if there is no errorMessage", () => {
+      renderFileUploader({
+        label: "Permit",
+        errorMessage: "",
+      });
+
+      expect(screen.getByText("Permit")).toBeInTheDocument();
+      expect(
+        screen.getByRole("button", { name: "Upload" })
+      ).toBeInTheDocument();
+      expect(screen.queryByRole("alert")).not.toBeInTheDocument();
+    });
+  });
+
   it('renders correctly when state is "error"', () => {
```

Comment:
Nit: Shouldn't this be part of the previous describe? Can we also update the test naming to reflect the difference in `fileName` being provided on whether it shows the upload button or the remove button?